### PR TITLE
refactor: error dialog uses reactive variables from apollo instead of react hooks

### PR DESCRIPTION
**Pull request by manuelblum**
- [x]  Implementation
- [x]  Changelog

### CODE REVIEW

**Review by kaufmo — APPROVED** [review]

File: packages/admin/admin/src/error/errordialog/ErrorDialogProvider.tsx [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
-            },
-        };
-    }, []);
+    React.useEffect(() => {
+        if (errorOptions && !errorDialogVisible) {
+            setErrorDialogVisible(true);
+        } else if (errorOptions == null && errorDialogVisible) {
+            setErrorDialogVisible(false);
+        }
+    }, [errorDialogVisible, errorOptions]);
```
Comment (johnnyomair) [inline packages/admin/admin/src/error/errordialog/ErrorDialogProvider.tsx:18]: I believe this useEffect can be greatly simplified:

```typescript
React.useEffect(() => {
    setErrorDialogVisible(errorOptions != null);
}, [errorOptions]);
```

Why would you need to check whether the error dialog is already visible? And setting the `errorDialogVisible` inside a useEffect that has `errorDialogVisible` as dependency may lead to infinite updates?

Comment (manuelblum) [reply packages/admin/admin/src/error/errordialog/ErrorDialogProvider.tsx:18]: true, thought tooooo complicated :-)

**Review by johnnyomair — APPROVED** [review]

**Review by kaufmo — APPROVED** [review]

**Comment by nsams** [comment]
good idea :+1:, I never liked the creating of apollo client inside the Provider component.

With this, ErrorDialog now has a dependency on apollo, but I don't see a problem with that.

`ErrorDialogProvider` isn't a provider anymore, and doesn't need to wrap the whole application anymore, should we change that?

**Comment by manuelblum** [comment]
> good idea 👍, I never liked the creating of apollo client inside the Provider component.
> 
> With this, ErrorDialog now has a dependency on apollo, but I don't see a problem with that.
> 
> `ErrorDialogProvider` isn't a provider anymore, and doesn't need to wrap the whole application anymore, should we change that?

The `ErrorDialogProvider` is not a classic Provider anymore when you mean `React.createContext`, but i think it is still necessary that it wrapps the whole application because the error dialog must life somewhere.

I am open for naming changes, i also thought to rename it, but i did't found a better naming for it. Do you have a renaming suggestion?

**Comment by nsams** [comment]
> but i think it is still necessary that it wrapps the whole application because the error dialog must life somewhere.

before that was required:

```
<ErrorDialogProvider>
    <TheWholeApplication />
</ErrorDialogProvider>
```

now that is possible:
```
<ErrorDialogProvider />
<TheWholeApplication />
```

**Comment by manuelblum** [comment]
sounds true and probably increases performance by reducing rerenders.

**Review by fraxachun — APPROVED** [review]

**Comment by nsams** [comment]
:+1:

Question remains if the "Provider" in the name is a good idea, as a user might think Providers always need to wrap. Just `ErrorDialog`? or `ErrorDialogHandler`? or `ErrorDialogManager`?

**Comment by manuelblum** [comment]
We can rename it to whatever we want. I would avoid naming it `ErrorDialog` because the UI Component is also named `ErrorDialog`, that may be confusing.

**Comment by johnnyomair** [comment]
> 👍
> 
> Question remains if the "Provider" in the name is a good idea, as a user might think Providers always need to wrap. Just `ErrorDialog`? or `ErrorDialogHandler`? or `ErrorDialogManager`?

I like `ErrorDialogHandler`. Another, even more explicit option may be `ErrorDialogRenderer`.

**Comment by nsams** [comment]
`ErrorDialogHandler` sounds good. (Renderer is imho too specific without the need for it)

**Comment by manuelblum** [comment]
renamed to `ErrorDialogHandler`

**Review by nsams — APPROVED** [review]

**Comment by johnnyomair** [comment]
@nsams remember remember, the last approver merges 😉

**Pull request merged**